nav2_bringup/launch/nav2_bringup_1st_launch.py

```diff
@@ -0,0 +1,84 @@
+
+# Copyright (c) 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import launch.actions
+from launch import LaunchDescription
+
+import launch_ros.actions
+
+from nav2_common.launch import RewrittenYaml
+
+
+def generate_launch_description():
+    map_yaml_file = launch.substitutions.LaunchConfiguration('map')
+    use_sim_time = launch.substitutions.LaunchConfiguration('use_sim_time',
+                                                            default='false')
+    autostart = launch.substitutions.LaunchConfiguration('autostart')
+    params_file = launch.substitutions.LaunchConfiguration('params')
+
+    # Create our own temporary YAML files that include substitutions
+    param_substitutions = {
+        'use_sim_time': use_sim_time,
+        'yaml_filename': map_yaml_file
+    }
+    configured_params = RewrittenYaml(
+        source_file=params_file, rewrites=param_substitutions,
+        convert_types=True)
+
+    return LaunchDescription([
+        launch.actions.SetEnvironmentVariable(
```

Comment:
Should we put a comment on why we need to set this Environment Variable?

Reply:
Sure, I can add that

Reply:
Done